src/Microsoft.DocAsCode.MarkdownLite/Gfm/GfmEmojiInlineRule.cs

```diff
@@ -0,0 +1,937 @@
+                ["clock1030"] = "🕥",
+                ["clock1130"] = "🕦",
+                ["heavy_multiplication_x"] = "✖️",
+                ["heavy_plus_sign"] = "➕",
+                ["heavy_minus_sign"] = "➖",
+                ["heavy_division_sign"] = "➗",
+                ["spades"] = "♠️",
+                ["hearts"] = "♥️",
+                ["clubs"] = "♣️",
+                ["diamonds"] = "♦️",
+                ["white_flower"] = "💮",
+                ["100"] = "💯",
+                ["heavy_check_mark"] = "✔️",
+                ["ballot_box_with_check"] = "☑️",
+                ["radio_button"] = "🔘",
+                ["link"] = "🔗",
+                ["curly_loop"] = "➰",
+                ["wavy_dash"] = "〰️",
+                ["part_alternation_mark"] = "〽️",
+                ["trident"] = "🔱",
+                ["black_medium_square"] = "◼️",
+                ["white_medium_square"] = "◻️",
+                ["black_medium_small_square"] = "◾",
+                ["white_medium_small_square"] = "◽",
+                ["black_small_square"] = "▪️",
+                ["white_small_square"] = "▫️",
+                ["small_red_triangle"] = "🔺",
+                ["black_square_button"] = "🔲",
+                ["white_square_button"] = "🔳",
+                ["black_circle"] = "⚫",
+                ["white_circle"] = "⚪",
+                ["red_circle"] = "🔴",
+                ["large_blue_circle"] = "🔵",
+                ["small_red_triangle_down"] = "🔻",
+                ["white_large_square"] = "⬜",
+                ["black_large_square"] = "⬛",
+                ["large_orange_diamond"] = "🔶",
+                ["large_blue_diamond"] = "🔷",
+                ["small_orange_diamond"] = "🔸",
+                ["small_blue_diamond"] = "🔹",
+                #endregion
+            };
+        }
+
+        public virtual string Name => "Inline.Gfm.Emoji";
+
+        public virtual Regex Emoji => Regexes.Inline.Gfm.Emoji;
+
+        protected virtual string GetEmoji(string shortCode)
+        {
+            string result;
+            _emoji.TryGetValue(shortCode, out result);
+            return result;
+        }
+
+        public virtual IMarkdownToken TryMatch(IMarkdownParser parser, IMarkdownParsingContext context)
+        {
+            var match = Emoji.Match(context.CurrentMarkdown);
+            if (match.Length == 0)
+            {
+                return null;
+            }
+            var shortCode = match.Groups[1].Value;
+            var text = GetEmoji(shortCode);
+            if (text == null)
+            {
+                var sourceInfo = context.Consume(1);
+                return new MarkdownTextToken(
+                    this,
+                    parser.Context,
+                    sourceInfo.Markdown,
+                    sourceInfo);
+            }
+            else
+            {
+                var sourceInfo = context.Consume(match.Length);
+                return new MarkdownTextToken(
```

Comment:
>MarkdownTextToken [](start = 27, length = 17)

this renders emoji as plain text. Is it better to render it as something like `<span class="smile">😄</span>` so that it's possible to do further customization through js/css?